.github/workflows/performance.yaml

```diff
@@ -0,0 +1,45 @@
+name: Performance Benchmark
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  hyperfine:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Cache dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: npm-${{ hashFiles('package-lock.json') }}
+          restore-keys: |
+            npm-${{ hashFiles('package-lock.json') }}
+            npm-
```

Comment:
no need to use it anymore, use setup-node@v3 action which has cache built-in and which is more convenient:

https://github.com/actions/setup-node#caching-global-packages-data